server/datastore/mysql/scripts.go

```diff
@@ -661,6 +663,28 @@ func (ds *Datastore) DeleteScript(ctx context.Context, id uint) error {
 			return ctxerr.Wrapf(ctx, err, "cancel pending script executions")
 		}
 
+		// load hosts that will have their upcoming_activities deleted, if that
+		// activity is "activated", as that means we will have to call
+		// activateNextUpcomingActivity for those hosts.
+		loadAffectedHostsStmt := `
+			SELECT
+				host_id
+			FROM
+				upcoming_activities ua
+				INNER JOIN script_upcoming_activities sua
+					ON ua.id = sua.upcoming_activity_id
+			WHERE sua.script_id = ? AND
+				ua.activity_type = 'script' AND
+				ua.activated_at IS NOT NULL AND
+				(ua.payload->'$.sync_request' = 0 OR
+					ua.created_at >= NOW() - INTERVAL ? SECOND)`
```

Comment:
Are there any additional indexes we'd want to add to support this query?

Reply:
This is already efficient as-is since its main condition is `script_id = ?` which is indexed (and of course the join with `ua` is also indexed).

Reply:
The JSON conditions are certainly less so, but I think it's a case of waiting to see if this becomes a problem, the delete has been there for a while with the same conditions without issue.